cloudify_nsx/library/nsx_security_policy.py

```diff
@@ -0,0 +1,227 @@
+
+    return "%s|%s" % (security_group_id, security_policy_id)
+
+
+def add_policy_section(client_session, security_policy_id, category, action):
+    raw_result = client_session.read(
+        'securityPolicyID', uri_parameters={'ID': security_policy_id}
+    )
+
+    common.check_raw_result(raw_result)
+
+    security_policy = raw_result['body']
+
+    actionsByCategory = security_policy['securityPolicy'].get(
+        'actionsByCategory', []
+    )
+
+    if isinstance(actionsByCategory, dict):
+        actionsByCategory = [actionsByCategory]
+
+    found = False
+    for actions in actionsByCategory:
+        if actions.get('category') == category:
+            found = True
+            actions['action'] = action
+
+    if not found:
+        actionsByCategory.append({
+            'category': category,
+            'action': action
+        })
+
+    security_policy['securityPolicy']['actionsByCategory'] = actionsByCategory
+
+    raw_result = client_session.update(
+        'securityPolicyID', uri_parameters={'ID': security_policy_id},
+        request_body_dict=security_policy
+    )
+
+    common.check_raw_result(raw_result)
+
+    return "%s|%s" % (category, security_policy_id)
+
+
+def del_policy_section(client_session, resource_id):
+    category, security_policy_id = resource_id.split("|")
+
+    raw_result = client_session.read(
+        'securityPolicyID', uri_parameters={'ID': security_policy_id}
+    )
+
+    common.check_raw_result(raw_result)
+
+    security_policy = raw_result['body']
+
+    actionsByCategory = security_policy['securityPolicy'].get(
+        'actionsByCategory', []
+    )
+
+    if isinstance(actionsByCategory, dict):
+        actionsByCategory = [actionsByCategory]
+
+    found = False
+    for actions in actionsByCategory:
+        if actions.get('category') == category:
+            actionsByCategory.remove(actions)
+            found = True
+            break
+
+    if not found:
+        return
+
+    security_policy['securityPolicy']['actionsByCategory'] = actionsByCategory
+
+    raw_result = client_session.update(
+        'securityPolicyID', uri_parameters={'ID': security_policy_id},
+        request_body_dict=security_policy
+    )
+
+    common.check_raw_result(raw_result)
+
+
+def del_policy_group_bind(client_session, resource_id):
+
+    security_group_id, security_policy_id = resource_id.split("|")
+
+    raw_result = client_session.read(
+        'securityPolicyID', uri_parameters={'ID': security_policy_id}
+    )
+
+    common.check_raw_result(raw_result)
+
+    security_policy = raw_result['body']
+
+    bindings = security_policy['securityPolicy'].get(
+        'securityGroupBinding', []
+    )
+
+    if isinstance(bindings, dict):
+        bindings = [bindings]
+
+    found = True
+    for bind in bindings:
+        if bind.get('objectId') == security_group_id:
+            bindings.remove(bind)
+            found = True
+            break
+
+    if not found:
```

Comment:
btw you can use `for/else` instead of setting the flag (which would help avoid such typos)